src/main/java/com/google/api/codegen/discogapic/DiscoGapicProvider.java

```diff
@@ -39,6 +43,17 @@ private DiscoGapicProvider(
     this.productConfig = productConfig;
     this.snippetSetRunner = snippetSetRunner;
     this.transformers = transformers;
+
+    ImmutableList.Builder<String> snippetFileNames = ImmutableList.builder();
+    for (DocumentToViewTransformer transformer : transformers) {
+      snippetFileNames.addAll(transformer.getTemplateFileNames());
+    }
+    this.snippetFileNames = snippetFileNames.build();
```

Comment:
For loops like this, consider
```java
this.snippetFilenames = transformers.stream().map(t -> t.getTemplateFileNames().stream()).collect(Collectors.asList());
```
Up to you.

Reply:
neato! google-cloud-java formatted the stream version in just as many LOC, so I'll leave it as old-school for-loop for now.